### PR TITLE
Testing Practices-JavaScript

### PR DESCRIPTION
**Pull request by gitnyasha**
Writing functions and testing with jest

### CODE REVIEW

File: array.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  const max = array.reduce((prev, cur) => (prev > cur ? prev : cur));
+  const { length } = array;
+
+  return {
+    average,
+    min,
+    max,
+    length,
+  };
+}
```
Comment (MauricioRobayo) [inline array.js:13]: You can also turn the whole function to an arrow function if you are into that kind of things, for example:

```js
const analyze = array => ({
  average: array.reduce((prev, cur) => prev + cur, 0) / array.length,
  min: array.reduce((prev, cur) => (prev < cur ? prev : cur)),
  max:  array.reduce((prev, cur) => (prev > cur ? prev : cur)),
  length: array.length,
})
```

File: array.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+test('returns minimum number', () => {
+  expect(analyze([1, 8, 3, 4, 2, 6]).min).toBe(1);
+});
+test('returns maximum number', () => {
+  expect(analyze([1, 8, 3, 4, 2, 6]).max).toBe(8);
+});
+
+test('calculates length', () => {
+  expect(analyze([1, 8, 3, 4, 2, 6]).length).toBe(6);
+});
```
Comment (MauricioRobayo) [inline array.test.js:16]: You could have a sinlge test for all of these:

```js
test('returns an object', () => {
  expect(analyze([1, 8, 3, 4, 2, 6])).toEqual({
    average: 4,
    min: 1,
    max: 8,
    length: 6,
  });
});
```
You could include some test to check what happens if an empty array is given, or if a value other than a number is given in the array.

File: capitalize.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
```
Comment (MauricioRobayo) [inline capitalize.js:3]: You can access elements in the string by index, and you could use an arrow function:

```js
const capitalize = string => string[0].toUpperCase() + string.slice(1)
```

**Review by MauricioRobayo — APPROVED** [review]
Nicely done, congratulations!

All tests green, all functions tested 👍 .

Status: __Approved__.

[OPTIONAL]: All comments on the code are optional. This is great work and it is fine the way it is 😄 🚀 .
